packages/admin/admin-rte/src/core/extension/NonBreakingSpace/EditorComponent.tsx

```diff
@@ -0,0 +1,23 @@
+import { styled } from "@mui/material/styles";
+import { ContentState } from "draft-js";
+import * as React from "react";
+
+interface Props {
+    contentState: ContentState;
+    entityKey: string;
+    children?: React.ReactNode;
+}
+
+function EditorComponent({ children }: Props): React.ReactElement {
+    return <VisibleHyphen>{children}</VisibleHyphen>;
+}
+
+const VisibleHyphen = styled("span")`
+    color: #999;
+    &:before {
+        margin-right: -4px; // Compensate for the non-breaking space that is added by the browser
```

Comment:
I doubt that this fixed 4px will be correct for all fonts and sizes. Maybe you can use rem unit? Or just ignore it